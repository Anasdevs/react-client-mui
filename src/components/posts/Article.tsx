import { Box, CardContent, Typography } from "@mui/material";
import { PostContent } from "../../types";

const Article = ({ title, text, native_text }: PostContent) => {
    return (
        <CardContent>
            <Typography mb={2} variant="h5" component="div">
                {title}
            </Typography>
            <Box>
                <Typography
                    mb={2}
                    style={{
                        display: "inline-block",
                        whiteSpace: "break-spaces",
                    }}
                >
                    {text}
                </Typography>
            </Box>
            <Box>
                <Typography
                    color="secondary"
                    style={{
                        display: "inline-block",
                        whiteSpace: "break-spaces",
                    }}
                >
                    {native_text}
                </Typography>
            </Box>
        </CardContent>
    );
};

export default Article;
