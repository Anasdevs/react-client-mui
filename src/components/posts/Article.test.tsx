import Article from "./Article";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import { it, expect } from "vitest";

it("renders without crashing", () => {
    render(<Article title="test" text="text" nativeText="native" />);
});

it("matches snapshot", () => {
    const { container } = render(
        <Article title="test" text="text" nativeText="native" />,
    );
    expect(container).toMatchSnapshot();
});
