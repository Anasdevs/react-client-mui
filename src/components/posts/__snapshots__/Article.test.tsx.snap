// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<div>
  <div
    class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <div
      class="MuiTypography-root MuiTypography-h5 css-ch2k2f-MuiTypography-root"
    >
      test
    </div>
    <div
      class="MuiBox-root css-0"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-z2eky3-MuiTypography-root"
        style="display: inline-block; white-space: break-spaces;"
      >
        text
      </p>
    </div>
    <div
      class="MuiBox-root css-0"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-18brw8f-MuiTypography-root"
        style="display: inline-block; white-space: break-spaces;"
      >
        native
      </p>
    </div>
  </div>
</div>
`;
